### ISSUE TITLE
Consider allowing hybrid clients to only request id_token

### ISSUE THREAD

**Issue by brockallen** [IdentityServer/IdentityServer4#2225]
Currently we fail the request.

**Comment by scottbrady91** [comment]
Is this allowed in the spec?

**Comment by brockallen** [comment]
The config for the client is orthogonal to the request from the client.

**Comment by scottbrady91** [comment]
I like the idea, but you're going to get people quote this table at you: https://openid.net/specs/openid-connect-core-1_0.html#Authentication

Also, playing devil's advocate, why just id_token? Couldn't hybrid client request all combinations?

**Comment by brockallen** [comment]
> I like the idea, but you're going to get people quote this table at you: https://openid.net/specs/openid-connect-core-1_0.html#Authentication

Yea, I see your point. 

> Also, playing devil's advocate, why just id_token? Couldn't hybrid client request all combinations?

Come to think of it, I like it. I only suggested id_token by itself since it reduces the complexity. The others need more consideration. 

The reason we have been so strict was to prevent misconfiguration. Granted, I only gave it a brief think-thru of threat models, but a hybrid client that also can just ask for id_token does not seen like it would leak more than code id_token. But I could see someone accidentally configuring hybrid when they only want id_token. Can that be abused? Possibly by an attacker sending code id_token then fooling the system into leaking a code, for example. But, I think our other 2 flags: allow access token via browser, and require client credentials for token requests can handle the potential abuses of unintended uses of those combinations.
